File: code/1040/1040.js
const input = require("fs").readFileSync("/dev/stdin", "utf8");
const lines = input.split(/\n|\s/);

const n1 = parseFloat(lines[0]) * 2;
const n2 = parseFloat(lines[1]) * 3;
const n3 = parseFloat(lines[2]) * 4;
const n4 = parseFloat(lines[3]);

const average = (n1 + n2 + n3 + n4) / 10;

console.log(`Media: ${average.toFixed(1)}`);

if (average < 5) {
  console.log("Aluno reprovado.");
} else if (average > 7) {
  console.log("Aluno aprovado.");
} else {
  const exam = parseFloat(lines[4]);
  console.log("Aluno em exame.");
  console.log(`Nota do exame: ${exam.toFixed(1)}`);
  console.log(
    (exam + average) / 2 < 5 ? "Aluno reprovado." : "Aluno aprovado."
  );
  console.log(`Media final: ${((exam + average) / 2).toFixed(1)}`);
}

File: code/1040/1040.cs
using System;

class URI
{
    static void Main(string[] args)
    {
        string[] input = Console.ReadLine().Split(' ');
        double score_1 = double.Parse(input[0]);
        double score_2 = double.Parse(input[1]);
        double score_3 = double.Parse(input[2]);
        double score_4 = double.Parse(input[3]);
        double average = (score_1 * 2 + score_2 * 3 + score_3 * 4 + score_4) / 10;
        average = Math.Round(average, 1);

        Console.WriteLine("Media: {0:0.0}", average);

        if (average >= 7.0)
        {
            Console.WriteLine("Aluno aprovado.");
        }
        else if (average < 5.0)
        {
            Console.WriteLine("Aluno reprovado.");
        }
        else
        {
            Console.WriteLine("Aluno em exame.");
            double exam_score = double.Parse(Console.ReadLine());
            Console.WriteLine("Nota do exame: {0:0.0}", exam_score);
            average = (average + exam_score) / 2;
            average = Math.Round(average, 1);

            if (average >= 5.0)
            {
                Console.WriteLine("Aluno aprovado.");
            }
            else
            {
                Console.WriteLine("Aluno reprovado.");
            }

            Console.WriteLine("Media final: {0:0.0}", average);
        }
    }
}
